flytekit/control_plane/launch_plan.py

```diff
@@ -0,0 +1,227 @@
+            default_inputs=_interface_models.ParameterMap(model.default_inputs.parameters),
+            fixed_inputs=model.fixed_inputs,
+            entity_metadata=model.entity_metadata,
+            labels=model.labels,
+            annotations=model.annotations,
+            auth_role=model.auth_role,
+            raw_output_data_config=model.raw_output_data_config,
+        )
+
+    @_exception_scopes.system_entry_point
+    def register(self, project, domain, name, version):
+        # NOTE: does this need to be implemented in the control plane?
+        pass
+
+    @classmethod
+    @_exception_scopes.system_entry_point
+    def fetch(cls, project: str, domain: str, name: str, version: str) -> "FlyteLaunchPlan":
+        """
+        This function uses the engine loader to call create a hydrated task from Admin.
+        :param project:
+        :param domain:
+        :param name:
+        :param version:
+        """
+        from flytekit.control_plane import workflow as _workflow
+
+        launch_plan_id = _identifier.Identifier(
+            _identifier_model.ResourceType.LAUNCH_PLAN, project, domain, name, version
+        )
+
+        lp = _flyte_engine.get_client().get_launch_plan(launch_plan_id)
+        flyte_lp = cls.promote_from_model(lp.spec)
+        flyte_lp._id = lp.id
+
+        # TODO: Add a test for this, and this function as a whole
+        wf_id = flyte_lp.workflow_id
+        lp_wf = _workflow.FlyteWorkflow.fetch(wf_id.project, wf_id.domain, wf_id.name, wf_id.version)
+        flyte_lp._interface = lp_wf.interface
+        return flyte_lp
+
+    @_exception_scopes.system_entry_point
+    def serialize(self):
+        """
+        Serializing a launch plan should produce an object similar to what the registration step produces,
+        in preparation for actual registration to Admin.
+
+        :rtype: flyteidl.admin.launch_plan_pb2.LaunchPlan
+        """
+        # NOTE: does this need to be implemented in the control plane?
+        pass
+
+    @property
+    def id(self) -> _identifier.Identifier:
+        return self._id
+
+    @property
+    def is_scheduled(self) -> bool:
+        if self.entity_metadata.schedule.cron_expression:
+            return True
+        elif self.entity_metadata.schedule.rate and self.entity_metadata.schedule.rate.value:
+            return True
+        elif self.entity_metadata.schedule.cron_schedule and self.entity_metadata.schedule.cron_schedule.schedule:
+            return True
+        else:
+            return False
+
+    @property
+    def auth_role(self) -> _common_models.AuthRole:
+        fixed_auth = super(FlyteLaunchPlan, self).auth_role
+        if fixed_auth is not None and (
+            fixed_auth.assumable_iam_role is not None or fixed_auth.kubernetes_service_account is not None
+        ):
+            return fixed_auth
+
+        assumable_iam_role = _auth_config.ASSUMABLE_IAM_ROLE.get()
```

Comment:
same here, we shouldn't modify what came from the database based on local settings.